astronomer/providers/apache/livy/triggers/livy.py

```diff
@@ -0,0 +1,133 @@
+                "extra_headers": self._extra_headers,
+                "livy_hook_async": self._livy_hook_async,
+            },
+        )
+
+    async def run(self) -> AsyncIterator["TriggerEvent"]:  # type: ignore[override]
+        """
+        Checks if the _polling_interval > 0, in that case it pools Livy for
+        batch termination asynchrnonously.
+        else returns the success response
+        """
+        try:
+            if self._polling_interval > 0:
+                response = await self.poll_for_termination(self._batch_id)
+                if response.get("status") == "success":
+                    yield TriggerEvent(response)
+                    return
+                else:
+                    yield TriggerEvent(response)
+                    return
+            yield TriggerEvent(
+                {
+                    "status": "success",
+                    "batch_id": self._batch_id,
+                    "response": f"Batch {self._batch_id} succeeded",
+                    "log_lines": None,
+                }
+            )
+        except Exception as exc:
+            yield TriggerEvent(
+                {
+                    "status": "error",
+                    "batch_id": self._batch_id,
+                    "response": f"Batch {self._batch_id} did not succeed with {str(exc)}",
+                    "log_lines": None,
+                }
+            )
+
+    async def poll_for_termination(self, batch_id: Union[int, str]) -> Dict[str, Any]:
+        """
+        Pool Livy for batch termination asynchrnonously.
+
+        :param batch_id: id of the batch session to monitor.
+        """
+        hook = self._get_async_hook()
+        state = await hook.get_batch_state(batch_id)
+        self.log.info("Batch with id %s is in state: %s", batch_id, state["batch_state"].value)
+        while state["batch_state"] not in hook.TERMINAL_STATES:
+            self.log.info("Batch with id %s is in state: %s", batch_id, state["batch_state"].value)
+            self.log.info("Sleeping for %s seconds", self._polling_interval)
+            await asyncio.sleep(self._polling_interval)
+            state = await hook.get_batch_state(batch_id)
+        self.log.info("Batch with id %s terminated with state: %s", batch_id, state["batch_state"].value)
+        log_lines = await hook.dump_batch_logs(batch_id)
+        if state["batch_state"] != BatchState.SUCCESS:
+            return {
+                "status": "error",
+                "batch_id": batch_id,
+                "response": f"Batch {batch_id} did not succeed",
+                "log_lines": log_lines,
+            }
+        return {
+            "status": "success",
+            "batch_id": batch_id,
+            "response": f"Batch {batch_id} succeeded",
+            "log_lines": log_lines,
+        }
+
+    def _get_async_hook(self) -> LivyHookAsync:
+        """
+        Get valid LivyHookAsync.
+        :return: hook
+        :rtype: LivyHookAsync
+        """
```

Comment:
Since this is an internal method, the description is same as method name.
```suggestion
    def _get_async_hook(self) -> LivyHookAsync:
```

Reply:
[Removed](https://github.com/astronomer/astronomer-providers/pull/178/commits/0ddab2aa4f20c5e780d0795e991c219f40beabde) it